blockchain/chain.go

```diff
@@ -1683,6 +1683,19 @@ func (b *BlockChain) MainChainHasBlock(hash *chainhash.Hash) bool {
 	return node != nil && b.bestChain.Contains(node)
 }
 
+// MedianTimeByHash returns the median time of a block by the given hash or zero
+// time if it doesn't exist.  Note that this will return times from both the
+// main chain and any side chains.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) MedianTimeByHash(hash *chainhash.Hash) time.Time {
```

Comment:
I'd rather this return an error rather than just silently failing when given a bogus hash.

Reply:
Returning error.